airflow/dags/unzip_and_validate_gtfs_schedule_hourly/unzip_gtfs_schedule.py

```diff
@@ -0,0 +1,282 @@
+    if pbar:
+        pbar.write(*args, **kwargs)
+    else:
+        typer.secho(*args, err=err, fg=fg, **kwargs)
+
+
+class GTFSScheduleFeedExtractUnzipOutcome(ProcessingOutcome):
+    extract: GTFSScheduleFeedExtract
+    zipfile_extract_md5hash: Optional[str]
+    zipfile_files: Optional[List[str]]
+    zipfile_dirs: Optional[List[str]]
+    extracted_files: Optional[List[GTFSScheduleFeedFile]]
+
+
+class ScheduleUnzipResult(PartitionedGCSArtifact):
+    bucket: ClassVar[str] = SCHEDULE_UNZIPPED_BUCKET
+    table: ClassVar[str] = "unzipping_results"
+    partition_names: ClassVar[List[str]] = ["dt", "ts"]
+    ts: pendulum.DateTime
+    outcomes: List[GTFSScheduleFeedExtractUnzipOutcome]
+
+    @property
+    def dt(self):
+        return self.ts.date()
+
+    @property
+    def successes(self) -> List[GTFSScheduleFeedExtractUnzipOutcome]:
+        return [outcome for outcome in self.outcomes if outcome.success]
+
+    @property
+    def failures(self) -> List[GTFSScheduleFeedExtractUnzipOutcome]:
+        return [outcome for outcome in self.outcomes if not outcome.success]
+
+    def save(self, fs):
+        self.save_content(
+            fs=fs,
+            content="\n".join(o.json() for o in self.outcomes).encode(),
+            exclude={"outcomes"},
+        )
+
+
+def summarize_zip_contents(
+    zip: zipfile.ZipFile,
+    at: str = "",
+    pbar=None,
+) -> Tuple[List[str], List[str], bool]:
+    files = []
+    directories = []
+    for entry in zip.namelist():
+        if zipfile.Path(zip, at=entry).is_file():
+            files.append(entry)
+        if zipfile.Path(zip, at=entry).is_dir():
+            directories.append(entry)
+    log(f"Found files: {files} and directories: {directories}", pbar=pbar)
+    # the only valid case for any directory inside the zipfile is if there's exactly one
+    # and it's the only item at the root of the zipfile(in which case we can treat that
+    # directory's contents as the feed)
+    # we want to exclude __MACOSX directories from this calculation
+    # see https://superuser.com/questions/104500/what-is-macosx-folder
+    is_valid = (not directories) or (
+        (len(directories) == 1)
+        and (
+            [
+                item.is_dir()
+                for item in zipfile.Path(zip, at="").iterdir()
+                if item.name != MACOSX_ZIP_FOLDER
+            ]
+            == [True]
+        )
+    )
+    return files, directories, is_valid
+
+
+def process_feed_files(
+    fs,
+    extract: GTFSScheduleFeedExtract,
+    zip: zipfile.ZipFile,
+    files: List[str],
+    directories: List[str],
+    is_valid: bool,
+    pbar=None,
+) -> Tuple[List[GTFSScheduleFeedFile], str]:
+    zipfile_files = []
+    md5hash = hashlib.md5()
+    if not is_valid:
+        raise ValueError(
+            "Unparseable zip: File/directory structure within zipfile cannot be unpacked"
+        )
+
+    # sorting is new here, to make the hash deterministic
```

Comment:
did you add this comment? can you just say "sorting makes the hash deterministic"? I don't like comments that are specific to the time context of the PR that added them, they will not make sense later